contracts/contracts/chains/ethereum/EthereumMessengers.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/access/Ownable.sol";
+
+import "../../../interfaces/IMessenger.sol";
+import "../../RestrictedCalls.sol";
+import "../../Resolver.sol";
+
+contract EthereumL1Messenger is IMessenger, RestrictedCalls {
+    function callAllowed(address, address) external view returns (bool) {
+        // The call must be delivered by the Resolver
+        // Since we are on the same chain as the Resolver
+        // we should never see an execution of this function.
+        revert("Unexpected state.");
+    }
+
+    // Simply forward the message to the target on L1
+    /* solhint-disable avoid-tx-origin */
+    function sendMessage(address target, bytes calldata message)
+        external
+        restricted(block.chainid)
+    {
+        target.call(message);
+    }
+}
+
+// In order to keep naming consistent between our rollup specific contracts
+// we are using `EthereumL2Messenger` here as a name even if this contract
+// is supposed to be deployed on L1.
+contract EthereumL2Messenger is IMessenger, Ownable, RestrictedCalls {
+    enum MessageStatus {
+        UNDEFINED,
+        DEFINED,
+        RELAYED
+    }
+
+    Resolver public immutable resolver;
+
+    /// Maps message hashes with their status.
+    mapping(bytes32 => MessageStatus) public messageHashes;
+
+    constructor(address resolver_) {
+        resolver = Resolver(resolver_);
+    }
+
+    function callAllowed(address, address) external view returns (bool) {
```

Comment:
not sure if we need to write this again as restrictedcalls seemed enough to me

Reply:
also true for L1Messenger part

Reply:
This is required as we are implementing the `IMessenger` interface.
The `callAllowed` plays an important role in the determining the trusted call chain.

In this case it just reverts because we are on the same chain and a call to this function should not be seen under normal operation of the protocol. 

This function is expected to be called here: https://github.com/beamer-bridge/beamer/blob/ethereum-messenger-contracts/contracts/contracts/RestrictedCalls.sol#L71

If you see the conditional statements you will notice that we should never reach that part of the code due to the fact that we are on the same chain (EthereumL1Messenger, EthereumL2Messenger, Resolver, FillManager are all on same chain in the "Ethereum as a rollup" case)

The code will always stop here: https://github.com/beamer-bridge/beamer/blob/ethereum-messenger-contracts/contracts/contracts/RestrictedCalls.sol#L63